modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala

```diff
@@ -0,0 +1,533 @@
+  }
+
+  private def statusIconRenderer(p: Props): CellRenderer[js.Object, js.Object, QueueRow] =
+    (_, _, _, row: QueueRow, _) => {
+      val icon: TagMod =
+        row.status match {
+          case SequenceState.Completed =>
+            IconCheckmark.copyIcon(fitted = true, extraStyles = List(SeqexecStyles.selectedIcon))
+          case SequenceState.Running(_, _) =>
+            IconCircleNotched.copyIcon(fitted = true,
+                                       loading = true,
+                                       extraStyles = List(SeqexecStyles.runningIcon))
+          case SequenceState.Failed(_) =>
+            IconAttention.copyIcon(fitted = true, extraStyles = List(SeqexecStyles.selectedIcon))
+          case _ =>
+            if (row.active)
+              IconSelectedRadio.copyIcon(fitted = true,
+                                         extraStyles = List(SeqexecStyles.selectedIcon))
+            else iconEmpty
+        }
+
+      val page = SequencePage(row.instrument, row.obsId, 0)
+      linkTo(p, page)(
+        SeqexecStyles.queueIconColumn,
+        icon
+      )
+    }
+
+  private val statusHeaderRenderer: HeaderRenderer[js.Object] = (_, _, _, _, _, _) =>
+    <.div(
+      ^.title := "Control",
+      ^.width := IconColumnWidth.px
+  )
+
+  // Renderer for a resizable column
+  private def resizableHeaderRenderer(
+      rs: (String, JsNumber) => Callback): HeaderRenderer[js.Object] =
+    (_, dataKey: String, _, label: VdomNode, _, _) =>
+      ReactFragment.withKey(dataKey)(
+        <.div(
+          ^.cls := "ReactVirtualized__Table__headerTruncatedText",
+          label
+        ),
+        Draggable(
+          Draggable.props(
+            axis = Axis.X,
+            defaultClassName = "DragHandle",
+            defaultClassNameDragging = "DragHandleActive",
+            onDrag = (ev: MouseEvent, d: DraggableData) => rs(dataKey, d.deltaX),
+            position = ControlPosition(0)
+          ),
+          <.span(^.cls := "DragHandleIcon", "⋮")
+        )
+    )
+
+  def rowClassName(p: Props)(i: Int): String =
+    ((i, p.rowGetter(i)) match {
+      case (-1, _) =>
+        SeqexecStyles.headerRowStyle
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s == SequenceState.Completed  =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowPositive
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s.isRunning                   =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowWarning
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s.isError                     =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowNegative
+      case (_, QueueRow(_, s, _, _, _, active, _)) if active && !s.isInProcess =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowActive
+      case _                                                                   =>
+        SeqexecStyles.stepRow
+    }).htmlClass
+
+  // scalastyle:off
+  private def columns($ : $, size: Size): List[Table.ColumnArg] = {
```

Comment:
I think I may have asked you this before a long time ago, but what does:

`$ : $`

mean?

Reply:
This is for historical convention though it is a bad name. `$` basically gives you access to the react `context`, including props and stat

Reply:
I could rename it to `backend` if it is clearer

Reply:
It's fine 👍 . I'll just take note of it.